Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp

```diff
@@ -1727,24 +1788,42 @@ void GStreamerMediaEndpoint::onDataChannel(GstWebRTCDataChannel* dataChannel)
     });
 }
 
-GstElement* GStreamerMediaEndpoint::requestAuxiliarySender()
+struct AuxiliarySenderDataHolder {
+    ThreadSafeWeakPtr<GStreamerMediaEndpoint> endPoint;
+    GRefPtr<GstWebRTCDTLSTransport> transport;
+};
+WEBKIT_DEFINE_ASYNC_DATA_STRUCT(AuxiliarySenderDataHolder)
+
+GstElement* GStreamerMediaEndpoint::requestAuxiliarySender(GRefPtr<GstWebRTCDTLSTransport>&& transport)
 {
     // Don't use makeGStreamerElement() here because it would be called mutiple times and emit an
     // error every single time if the element is not found.
     auto* estimator = gst_element_factory_make("rtpgccbwe", nullptr);
     if (!estimator) {
         static std::once_flag onceFlag;
         std::call_once(onceFlag, [] {
-            WTFLogAlways("gst-plugins-rs is not installed, RTP bandwidth estimation now disabled");
+            gst_printerrln("gst-plugins-rs is not installed, RTP bandwidth estimation now disabled");
         });
         return nullptr;
     }
 
-    g_signal_connect(estimator, "notify::estimated-bitrate", G_CALLBACK(+[](GstElement* estimator, GParamSpec*, GStreamerMediaEndpoint* endPoint) {
+    auto holder = createAuxiliarySenderDataHolder();
+    holder->endPoint = this;
+    holder->transport = WTFMove(transport);
+
+    g_signal_connect_data(estimator, "notify::estimated-bitrate", G_CALLBACK(+[](GstElement* estimator, GParamSpec*, gpointer userData) {
+        auto holder = static_cast<AuxiliarySenderDataHolder*>(userData);
+        RefPtr endPoint = holder->endPoint.get();
+        if (!endPoint)
+            return;
+
         uint32_t estimatedBitrate;
         g_object_get(estimator, "estimated-bitrate", &estimatedBitrate, nullptr);
-        gst_element_send_event(endPoint->pipeline(), gst_event_new_custom(GST_EVENT_CUSTOM_DOWNSTREAM_OOB, gst_structure_new("encoder-bitrate-change-request", "bitrate", G_TYPE_UINT, static_cast<uint32_t>(estimatedBitrate / 1000), nullptr)));
-    }), this);
+
+        endPoint->m_peerConnectionBackend.dispatchSenderBitrateRequest(holder->transport, estimatedBitrate);
+    }), holder, reinterpret_cast<GClosureNotify>(+[](gpointer data, GClosure*) {
+        destroyAuxiliarySenderDataHolder(static_cast<AuxiliarySenderDataHolder*>(data));
+    }), static_cast<GConnectFlags>(0));
```

Comment:
I see that 0 is `G_CONNECT_DEFAULT` (https://docs.gtk.org/gobject/flags.ConnectFlags.html#default), but that all WebKit code consistently does `static_cast<GConnectFlags>(0)`.

Is there a reason why we're not using the named constant?